File: api/prompts/characters.py
"""Coach persona templates — tone and style only; guardrails always apply separately."""

from typing import Final

CharacterId = str

CHARACTERS: Final[dict[CharacterId, str]] = {
    "professional": (
        "Persona: Professional coach. "
        "Speak in a clear, structured, and composed tone. "
        "Use organized responses with practical steps when helpful. "
        "Be warm but measured — like a skilled workplace or life coach."
    ),
    "warm_friend": (
        "Persona: Warm friend. "
        "Speak in a conversational, empathetic, and validating tone. "
        "Acknowledge feelings first, use everyday language, and make the user feel heard. "
        "Be supportive without being overly clinical."
    ),
    "mindful_guide": (
        "Persona: Mindful guide. "
        "Speak in a calm, reflective, and grounding tone. "
        "Encourage present-moment awareness, gentle breathing, and self-compassion. "
        "Use unhurried language and thoughtful pauses in your phrasing."
    ),
    "motivational_coach": (
        "Persona: Motivational coach. "
        "Speak in an encouraging, energizing, and action-oriented tone. "
        "Highlight strengths, celebrate small wins, and help the user identify concrete next steps. "
        "Stay optimistic and empowering without being dismissive of difficult feelings."
    ),
}

DEFAULT_CHARACTER: CharacterId = "mindful_guide"

VALID_CHARACTERS: frozenset[CharacterId] = frozenset(CHARACTERS.keys())


def get_character_prompt(character: CharacterId) -> str:
    """Return the persona prompt for a valid character id."""
    if character not in CHARACTERS:
        raise ValueError(f"Unknown character: {character}")
    return CHARACTERS[character]

File: frontend/src/lib/characters.ts
/** Supported coach persona identifiers — must match the backend. */
export type CoachCharacterId =
  | "professional"
  | "warm_friend"
  | "mindful_guide"
  | "motivational_coach";

export interface CoachCharacter {
  id: CoachCharacterId;
  label: string;
  description: string;
  icon: string;
}

export const COACH_CHARACTERS: CoachCharacter[] = [
  {
    id: "professional",
    label: "Professional",
    description: "Clear and structured",
    icon: "💼",
  },
  {
    id: "warm_friend",
    label: "Warm friend",
    description: "Empathetic and validating",
    icon: "🤗",
  },
  {
    id: "mindful_guide",
    label: "Mindful guide",
    description: "Calm and reflective",
    icon: "🧘",
  },
  {
    id: "motivational_coach",
    label: "Motivational coach",
    description: "Encouraging and action-oriented",
    icon: "⚡",
  },
];

export const CHARACTER_STORAGE_KEY = "mindful-coach-character";

export const DEFAULT_CHARACTER: CoachCharacterId = "mindful_guide";

const VALID_IDS = new Set<string>(COACH_CHARACTERS.map((c) => c.id));

export function isCoachCharacterId(value: string): value is CoachCharacterId {
  return VALID_IDS.has(value);
}

export function readStoredCharacter(): CoachCharacterId {
  if (typeof window === "undefined") {
    return DEFAULT_CHARACTER;
  }

  const stored = localStorage.getItem(CHARACTER_STORAGE_KEY);
  if (stored && isCoachCharacterId(stored)) {
    return stored;
  }

  return DEFAULT_CHARACTER;
}

export function getCharacterById(id: CoachCharacterId): CoachCharacter {
  const character = COACH_CHARACTERS.find((c) => c.id === id);
  if (!character) {
    throw new Error(`Unknown character: ${id}`);
  }
  return character;
}
